crates/shared-crypto/src/intent.rs

```diff
@@ -170,10 +170,12 @@ pub(crate) mod private {
     impl<T> SealedIntent for IntentMessage<T> {}
 }
 
-/// Hashing intent is used to prevent hash collision when hashing a message.
+/// A 1-byte domain separator for hashing Object ID in Sui. It is starting from 0xf0
+/// to ensure no hashing collision for any ObjectID vs SuiAddress which is derived
+/// as the hash of `flag || pubkey`. See [`sui_types::crypto::SignatureScheme::flag()`].
```

Comment:
we can avoid the link here (add it as simple string so CI doesn't complain), and we can fix cross-dep doc links in upcoming DOCathons.